File: frontend/src/types/faultAnalysis.ts
export type Severity = 'Routine' | 'NeedsAttention' | 'SafetyCritical';

export interface FaultAnalysis {
  summary: string;
  affectedSystems: string[];
  severity: Severity;
  clarifyingQuestions: string[];
  suggestedNextSteps: string[];
}

File: backend/FaultTriage.Core/FaultAnalysis.cs
namespace FaultTriage.Core;

public record FaultAnalysis(
    string Summary,
    IReadOnlyList<string> AffectedSystems,
    Severity Severity,
    IReadOnlyList<string> ClarifyingQuestions,
    IReadOnlyList<string> SuggestedNextSteps
);